### ISSUE TITLE
naming a enum Literal "false" fails in Xtext2Ecore & Ecore2Xtext

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#1547]
[XtextSyntaxDiagnostic: null:955 mismatched input 'false' expecting RULE_ID, XtextSyntaxDiagnostic: null:955 mismatched input '=' expecting ':', XtextSyntaxDiagnostic: null:955 mismatched input '=' expecting ':', XtextSyntaxDiagnostic: null:955 mismatched input '=' expecting ':', TransformationDiagnostic: null:955 Cannot create type for false because its EPackage is unknown. (ErrorCode: UnknownMetaModelAlias), TransformationDiagnostic: null:955 Cannot create type for partial because its EPackage is unknown. (ErrorCode: UnknownMetaModelAlias), TransformationDiagnostic: null:955 Cannot create type for total because its EPackage is unknown. (ErrorCode: UnknownMetaModelAlias)]

```
enum XSDOrdered returns xsd::XSDOrdered:
				false = 'false' | partial = 'partial' | total = 'total';
```

**Comment by cdietrich** [comment]
looks like the false= is not escaped.
any idea @szarnekow 
arent conditions a pretty old feature?

**Comment by cdietrich** [comment]
hmmm even escaping does not work.
at least with inference

"name is not well formed"

**Comment by cdietrich** [comment]
guess it should be 

EnumLiteralDeclaration:
	enumLiteral=[ecore::EEnumLiteral|ValidID] ('=' literal=Keyword)?
;

**Comment by szarnekow** [comment]
Xtext.xtext: `enumLiteral=[ecore::EEnumLiteral]` should probably be `enumLiteral=[ecore::EEnumLiteral|ValidID]`

**Comment by cdietrich** [comment]
this does not seem to work unfortunately

**Comment by cdietrich** [comment]
the value is

```
\n\t\t\t\t^false
```
so trimming does not seem to work there

**Comment by szarnekow** [comment]
Which value is `\n\t\t\t\t^false` ?

**Comment by cdietrich** [comment]
the one that goes to the value converter
org.eclipse.xtext.conversion.impl.KeywordAlternativeConverter.toString(String)

**Comment by cdietrich** [comment]
see org.eclipse.xtext.xtext.ecoreInference.Xtext2EcoreTransformer.deriveEnums(EnumRule)

**Comment by cdietrich** [comment]
for now i just have a dirty fix.the clean solution would be to use value converters everywhere i guess

**Comment by cdietrich** [comment]
we will go with the lightweight solution for now.
will create follow up issue for later

https://github.com/eclipse/xtext/issues/2496

**Comment by cdietrich** [comment]
fixed in 2.24
